flink-core/src/test/java/org/apache/flink/core/io/PostVersionedIOReadableWritableTest.java

```diff
@@ -35,52 +36,92 @@
 
 	@Test
 	public void testReadVersioned() throws IOException {
+		byte[] payload = "test-data".getBytes();
+		byte[] serialized = serializeWithPostVersionedReadableWritable(payload);
+		byte[] restored = restoreWithPostVersionedReadableWritable(serialized, payload.length);
 
-		String payload = "test-data";
-		TestPostVersionedReadableWritable versionedReadableWritable = new TestPostVersionedReadableWritable(payload);
+		Assert.assertArrayEquals(payload, restored);
+	}
+
+	@Test
+	public void testReadNonVersioned() throws IOException {
+		byte[] preVersionedPayload = new byte[]{0x00, 0x00, 0x02, 0x33};
+		byte[] serialized = serializeWithNonVersionedReadableWritable(preVersionedPayload);
+		byte[] restored = restoreWithPostVersionedReadableWritable(serialized, preVersionedPayload.length);
+
+		Assert.assertArrayEquals(preVersionedPayload, restored);
+	}
+
+	@Test
+	public void testReadNonVersionedWithLongPayload() throws IOException {
+		byte[] preVersionedPayload = "test-data".getBytes();
+		byte[] serialized = serializeWithNonVersionedReadableWritable(preVersionedPayload);
+		byte[] restored = restoreWithPostVersionedReadableWritable(serialized, preVersionedPayload.length);
+
+		Assert.assertArrayEquals(preVersionedPayload, restored);
+	}
+
+	@Test
+	public void testReadNonVersionedWithShortPayload() throws IOException {
```

Comment:
Can you add one more UT, where the pre-versioned payload is empty?
I don't expect it to fail, but would be nice to have.

Reply:
Thanks for reviewing. Good catch as empty payload test reveals an issue around uses of PushbackInputStream. Fixed in new revision.